src/NuGet.Core/NuGet.Protocol.Core.v3/CachingSourceProvider.cs

```diff
@@ -22,7 +22,7 @@ public class CachingSourceProvider : ISourceRepositoryProvider
 
         // There should only be one instance of the source repository for each package source.
         private readonly ConcurrentDictionary<string, SourceRepository> _cachedSources
-            = new ConcurrentDictionary<string, SourceRepository>(StringComparer.OrdinalIgnoreCase);
+            = new ConcurrentDictionary<string, SourceRepository>(StringComparer.Ordinal);
```

Comment:
This came from a comment in PR 586: sources should be cached in a case-sensitive dictionary